application/frontend/controllers/MfaController.php

```diff
@@ -140,17 +140,40 @@ public function actionDelete($mfaId)
 
     /**
      * @param $mfaId
-     * @return array|bool
-     * @throws HttpException
+     * @param $webauthnId
+     * @return null
+     * @throws ServiceException
+     * @throws NotFoundHttpException
      */
-    public function actionVerify($mfaId)
+    public function actionDeletewebauthn($mfaId, $webauthnId)
     {
-        $messages = [
-            400 => \Yii::t('app', 'Mfa.InvalidCode'),
-            404 => \Yii::t('app', 'Mfa.VerifyFailure'),
-            429 => \Yii::t('app', 'Mfa.RateLimitFailure'),
-        ];
+        try {
+            return $this->idBrokerClient->mfaDeleteWebauthn(
+                $mfaId,
+                \Yii::$app->user->identity->employee_id,
+                $webauthnId
+            );
+        } catch (ServiceException $e) {
+            \Yii::error([
+                'status' => 'MFA delete webauthn error',
+                'message' => $e->getMessage(),
+            ], __METHOD__);
+            if ($e->httpStatusCode == 404) {
+                throw new NotFoundHttpException(\Yii::t('app', 'Mfa.RecordNotFound'));
+            }
+
+            /*
+             * Other status codes will result in a 500 response
+             */
+            throw $e;
+        }
+    }
 
+    /**
+     * @return array
+     * @throws HttpException
+     */
+    private static function getVerifyValue() {
         $value = \Yii::$app->request->getBodyParam('value');
         if ($value === null) {
             throw new BadRequestHttpException(\Yii::t('app', 'Mfa.MissingValue'));
```

Comment:
Now that this is in a separate private method (rather than in the controller action), it would probably be better to simply let this `getVerifyValue()` method return `null`, and leave the calling method (the actual controller action that's handling the request) throw the `BadRequestHttpException` if the value is `null`.
```suggestion
        if ($value === null) {
            return null;
```

Reply:
I don't mind changing it, but I'm confused why you would want an `if` block repeated in more than one calling method when you can have it just once in the called method.

Reply:
You made a good point. As long as the HTTP exception is still in the controller (which it is in both of these options), it's still in a good location. Thanks for pushing back on that... you're right.